Proposal for senior-software-backend-engineer position (#11546)
fleetdm/fleet: handbook/company/senior-software-backend-engineer.md

- 🤝 Collaboration: You work best in a participatory, team-based environment.
- 🚀 Prototype-first: You embrace speed and failure as we iterate towards the right solution. You have hands-on experience in creating low and high fidelity prototypes. You’re comfortable accepting suboptimal designs in favor of iteration.
- 🧬 Simplicity: You love complex questions and use your work to simplify that complexity for users.
- 🛠️ Technical: You understand the software development processes. You understand that software quality matters.
- 🟣 Openness: You are flexible and open to new ideas and ways of working.
- ➕ Bonus: Cybersecurity or IT background.

### You can "talk the talk":

- 💭 3-5 years' of experience in backend/SaaS development.
- 🦉 Proficient in backend development. You practice OOP design and are comfortable in a lean software development environment. 

## Why should you join us? 🛸

Learn more about the company and [why you should join us here](https://fleetdm.com/handbook/company#is-it-any-good).

## Want to join the team?

You can [apply for this position here](https://3x3q33auqgj.typeform.com/to/upGkhYsN).

<meta name="maintainedBy" value="mikermcneil">

> <meta name="title" value="🚀 Product Designer">

Reviewer: ```suggestion
<meta name="title" value="🚀 Senior Software Engineer (Golang)">
```

Reviewer: Noticed one last thing.

> # 🚀 Senior Software Backend Engineer

Reviewer: ```suggestion
# 🚀 Senior Software Engineer (Golang)
```

- 🦉 Translate requirements into well-designed and functional software.
- 🤝 Communicate regularly with stakeholders, project managers, quality assurance teams, and other developers regarding progress on long-term technology roadmap.
- 🧪 Collaborate with QA team for testing software features.
- 🏃‍♂️ Familiarity with agile development processes and scrum methodologies.
- 🛠️ Produce quality code, raising the bar for team performance and speed.
- 📖 Mentor junior team members.

### You can "walk the walk":

- 🤝 Collaboration: You work best in a participatory, team-based environment.
- 🚀 Prototype-first: You embrace speed and failure as we iterate towards the right solution. You have hands-on experience in creating low and high fidelity prototypes. You’re comfortable accepting suboptimal designs in favor of iteration.
- 🧬 Simplicity: You love complex questions and use your work to simplify that complexity for users.
- 🛠️ Technical: You understand the software development processes. You understand that software quality matters.
- 🟣 Openness: You are flexible and open to new ideas and ways of working.
- ➕ Bonus: Cybersecurity or IT background.

### You can "talk the talk":

> - 📖 Bachelor’s degree (or equivalent) in computer science

Reviewer: ```suggestion
```

Reviewer: We don't list this as a requirement to avoid discouraging quality engineers without degrees.

- 💭 3-5 years' of experience in backend/SaaS development.

> - 🦉 You are proficient in software backend development. You practice OOP design, and are comfortable in a lean software development environment. 

Reviewer: ```suggestion
- 🦉 Proficient in backend development. You practice OOP design and are comfortable in a lean software development environment. 
```

Organizations make huge investments every year to keep their laptops and servers online, secure, compliant, and usable from anywhere. This is called "device management".

At Fleet, we think it's time device management became [transparent](https://fleetdm.com/transparency) and [open source](https://fleetdm.com/handbook/company#open-source).

## About the company 🌈

You can read more about the company in our [handbook](https://fleetdm.com/handbook/company), which is public and open to the world.

tldr; Fleet Device Management Inc. is a [recently-funded](https://techcrunch.com/2022/04/28/fleet-nabs-20m-to-enable-enterprises-to-manage-their-devices/) Series A startup founded and backed by the same people who created osquery, the leading open source security agent. Today, osquery is installed on millions of laptops and servers, and it is especially popular with [enterprise IT and security teams](https://www.linuxfoundation.org/press/press-release/the-linux-foundation-announces-intent-to-form-new-foundation-to-support-osquery-community).

### Your primary responsibilities 🔍

As a Senior Software Engineer at Fleet, you will get the chance to…

In your first 90 days:

> - 🧑‍🔬 Design, Develop, Test and maintain a state-of-the-art Backend code including robust APIs to support mobile and desktop clients.

Reviewer: ```suggestion
- 🧑‍🔬 Design, develop, test, and maintain a state-of-the-art Golang application that includes robust APIs to support mobile and desktop clients.
```

- 🛠️ Write code and tests, build prototypes, resolve issues, and profile and analyze bottlenecks.

> - 💭 Manage and optimize scalable distributed systems in the cloud

Reviewer: ```suggestion
- 💭 Manage and optimize scalable distributed systems in the cloud.
```

> - 🤝 Collaborate closely with Product managers to understand requirements and translate them into work plans and software design.

Reviewer: ```suggestion
- 🤝 Collaborate closely with product managers to understand requirements and translate them into actionable specifications.
```

> - 🚀 Actively participating in all engineering scrum meetings, including sprint planning, daily standups, sprint demos, sprint retrospectives, and estimation sessions.

Reviewer: ```suggestion
- 🚀 Actively participate in all engineering scrum meetings, including sprint planning, daily standups, sprint demos, sprint retrospectives, and estimation sessions.
```

- 🌟 Contribute to the overall success of the [customer cxperience (CX)](https://fleetdm.com/handbook/company/development-groups#customer-experience-group) product group by ensuring users receive valuable new features.

### Are you our new team member? 🧑‍🚀

If most of these qualities sound like you, we would love to chat and see if we're a good fit.

### You "get it":

> - 🦉 Translate requirements into proper design and software.

Reviewer: ```suggestion
- 🦉 Translate requirements into well-designed and functional software.
```

- 🤝 Communicate regularly with stakeholders, project managers, quality assurance teams, and other developers regarding progress on long-term technology roadmap.

> - 🧪 Develop and oversee testing schedules in client server environment.

Reviewer: I'm not sure what this means. Clarify?

- 🏃‍♂️ Familiarity with agile development processes and scrum methodologies.
- 🛠️ Produce quality code, raising the bar for team performance and speed.

> - 📖 Mentor Junior team members.

Reviewer: ```suggestion
- 📖 Mentor junior team members.
```

### You can "walk the walk":

- 🤝 Collaboration: You work best in a participatory, team-based environment.

> - 📈 Produce quality code, raising the bar for team performance and speed

Reviewer: Duplicate of an item in the previous list. Please remove one of them.

- 🚀 Prototype-first: You embrace speed and failure as we iterate towards the right solution. You have hands-on experience in creating low and high fidelity prototypes. You’re comfortable accepting suboptimal designs in favor of iteration.
- 🧬 Simplicity: You love complex questions and use your work to simplify that complexity for users.
- 🛠️ Technical: You understand the software development processes. You understand that software quality matters.
- 🟣 Openness: You are flexible and open to new ideas and ways of working.
- ➕ Bonus: Cybersecurity or IT background.

### You can "talk the talk":

- 📖 Bachelor’s degree (or equivalent) in computer science
- 💭 3-5 years' of experience in backend/SaaS development.

> - 🦉 You are Proficient in software backend development. You practice OOP design, design patterns and SOLID principles.

Reviewer: ```suggestion
- 🦉 You are proficient in software backend development. You practice OOP design, and are comfortable in a lean software development environment. 
```

> - 🦉 You are Proficient in software backend development. You practice OOP design, design patterns and SOLID principles.

Reviewer: I don't mind listing SOLID principles, but we should include something about them in the handbook first if we require it. We talk about agile, scrum, and lean software development in the handbook, so I'm suggesting that instead.

Author: SGTM 👍
